addons/product/models/product.py

```diff
@@ -378,17 +378,37 @@ def _name_get(d):
         self.check_access_rule("read")
 
         result = []
+
+        # Prefetch the fields used by the `name_get`, so `browse` doesn't fetch other fields
+        # Use `load=False` to not call `name_get` for the `product_tmpl_id`
+        self.sudo().read(['name', 'default_code', 'product_tmpl_id', 'attribute_value_ids'], load=False)
+
+        product_template_ids = [product.product_tmpl_id.id for product in self]
```

Comment:
Would be better to be a list, to remove duplicate product_templates.
product_template_ids = list([product.product_tmpl_id.id for product in self])

Reply:
Set sorry , at least that was my suggestion

Reply:
Good point. I would try `self.mapped('product_tmpl_id').ids`

Reply:
Good spot, have now changed this. Also added sudo call to make use of the prefetch for `product_tmpl_id`.